crates/countersyncd/src/actor/otel.rs

```diff
@@ -0,0 +1,294 @@
+        }
+    }
+}
+
+/// Actor that receives SAI statistics and exports to OpenTelemetry
+pub struct OtelActor {
+    stats_receiver: Receiver<SAIStatsMessage>,
+    config: OtelActorConfig,
+    shutdown_notifier: Option<oneshot::Sender<()>>,
+    client: MetricsServiceClient<tonic::transport::Channel>,
+
+    // Statistics tracking
+    messages_received: u64,
+    exports_performed: u64,
+    export_failures: u64,
+    console_reports: u64,
+}
+
+impl OtelActor {
+    /// Creates a new OtelActor instance
+    pub async fn new(
+        stats_receiver: Receiver<SAIStatsMessage>,
+        config: OtelActorConfig,
+        shutdown_notifier: oneshot::Sender<()>
+    ) -> Result<OtelActor, Box<dyn std::error::Error>> {
+        let endpoint = config.collector_endpoint.parse::<Endpoint>()?;
+        let client = MetricsServiceClient::connect(endpoint).await?;
+
+        info!(
+            "OtelActor initialized - console: {}, endpoint: {}",
+            config.print_to_console,
+            config.collector_endpoint
+        );
+
+        Ok(OtelActor {
+            stats_receiver,
+            config,
+            shutdown_notifier: Some(shutdown_notifier),
+            client,
+            messages_received: 0,
+            exports_performed: 0,
+            export_failures: 0,
+            console_reports: 0,
+        })
+    }
+
+    /// Main run loop
+    pub async fn run(mut self) {
+        info!("OtelActor started");
+
+        loop {
+            select! {
+                stats_msg = self.stats_receiver.recv() => {
+                    match stats_msg {
+                        Some(stats) => {
+                            self.handle_stats_message(stats).await;
+                        }
+                        None => {
+                            info!("Stats receiver channel closed, shutting down OtelActor");
+                            break;
+                        }
+                    }
+                }
+            }
+        }
+
+        self.shutdown().await;
+    }
+
+    /// Handle incoming SAI statistics message
+    async fn handle_stats_message(&mut self, stats: SAIStatsMessage) {
+        self.messages_received += 1;
+
+        debug!("Received SAI stats with {} entries, observation_time: {}",
+               stats.stats.len(), stats.observation_time);
+
+        // Convert to OTel format using message types
+        let otel_metrics = OtelMetrics::from_sai_stats(&stats);
+
+        // Print to console if enabled
+        if self.config.print_to_console {
+            self.print_otel_metrics(&otel_metrics).await;
+        }
+
+        // Export to OpenTelemetry collector
+        self.export_otel_metrics(&otel_metrics).await;
+    }
+
+    /// Print metrics to console
+    async fn print_otel_metrics(&mut self, otel_metrics: &OtelMetrics) {
+        self.console_reports += 1;
+
+        println!("   [OTel Report #{}] OpenTelemetry Metrics Export", self.console_reports);
```

Comment:
I suggest don't directly use println to print any information to console, it's better to use log: https://github.com/sonic-net/sonic-swss/blob/3ccfa6299c8722e528cd5eba50805f753c2b3177/crates/countersyncd/src/actor/stats_reporter.rs#L5